R/checkSummations.R

```diff
@@ -84,8 +84,9 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = NULL, sum
     if (isTRUE(summationsFile == "extractVariableGroups")) {
       comp$factor <- 1
     } else {
-      comp <- left_join(comp, select(summationGroups, c("child", "factor")),
-                        by = c("child"), relationship = "many-to-many")
+      summation <- filter(summationGroups, parent == names(checkVariables[i]))
+      comp <- left_join(comp, summation, by = c("child", "variable" = "parent"),
+                        relationship = "many-to-many")
```

Comment:
do you need the "many-to-many" still? I don't think so, because you should rather not have the same child multiple times for a specific parent, right? So I think that would rather show an error in the summation file and therefore should raise a warning…